sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        """
+        return self._boundary_conditions
+
+    def apply_boundary_conditions(self, moment = [], slope = [], deflection = []):
+        """
+        Takes the boundary conditions of the beam bending problem as input.
+        The boundary conditions should be passed as keyworded arguments.
+        It is suggested to use ``moment``, ``slope`` and ``deflection`` as the
+        keywords while providing the boundary conditions of the moment curve,
+        slope curve and the deflection curve respectively as inputs.
+        A tuple in the form of (location, value) can be used to pass the
+        location and the value of a boundary condition. A list of such tuples
+        is passed under each keywords.
+
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+
+        Here ``moment = [(0, 4), (4, 0)]`` means the moment due to the
+        applied loads at point ``0`` and ``4`` are `4` and `0` respectively.
+        Similarly, ``deflection = [(0, 2)]`` means the deflection of the beam
+        at point ``0`` is ``2``.
+
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0)]
+
+        """
+        if moment:
+            self._boundary_conditions['moment'].extend([m_bcs for m_bcs in moment])
+        if slope:
+            self._boundary_conditions['slope'].extend([s_bcs for s_bcs in slope])
+        if deflection:
+            self._boundary_conditions['deflection'].extend([d_bcs for d_bcs in deflection])
+
+    def apply_moment_boundary_conditions(self, *m_bcs):
+        """
+        Takes only the moment boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_moment_boundary_conditions((4, 3), (5, 0))
```

Comment:
What is the reason for having both `apply_boundary_conditions` and `apply_moment_boundary_conditions`? Seems redundant.